### PR TITLE
fix: handle attribute collision

### PR DESCRIPTION
**Pull request by tushar-signoz**
### 📄 Summary
This pull request improves the handling of field key resolution in query building and updates the VSCode settings for Go development. The main focus is to make the field key matching logic in the `VisitKey` function more robust, especially when dealing with context and data type prefixes.

Field key resolution improvements:

* Refactored the logic in `VisitKey` (`pkg/querybuilder/where_clause_visitor.go`) to more accurately filter field keys by context and data type, ensuring that only relevant keys are considered. Now, keys are matched if their context and data type are either unspecified or explicitly match, and additional logic handles cases where the context might actually be part of the name (e.g., `span.div_num`).
* Removed redundant fallback logic that attempted to resolve keys by combining context and name, as this is now handled in the main matching logic.

Development environment configuration:

* Updated `.vscode/settings.json` to set the default code formatter for Go files to `golang.go`, improving the Go development experience in VSCode.

---

### ✅ Change Type
_Select all that apply_

- [ ] ✨ Feature
- [ ] 🐛 Bug fix
- [ ] ♻️ Refactor
- [ ] 🛠️ Infra / Tooling
- [ ] 🧪 Test-only

---

---

- Tests added/updated:
- Manual verification:
- Edge cases covered:

---

- Blast radius:
- Potential regressions:
- Rollback plan:

---

| Field | Value |
|------|-------|
| Deployment Type | Cloud / OSS / Enterprise |
| Change Type | Feature / Bug Fix / Maintenance |
| Description | User-facing summary |

---

### 📋 Checklist
- [ ] Tests added or explicitly not required
- [ ] Manually tested
- [ ] Breaking changes documented
- [ ] Backward compatibility considered

---

## 👀 Notes for Reviewers

---

### CODE REVIEW

File: pkg/querybuilder/where_clause_visitor.go [hunk 1]
```diff
@@ -809,32 +809,35 @@ func (v *filterExpressionVisitor) VisitKey(ctx *grammar.KeyContext) any {
...
+
+	if fieldKey.FieldContext != telemetrytypes.FieldContextUnspecified {
+
+		contextPrefixedFieldName := fmt.Sprintf("%s.%s", fieldKey.FieldContext.StringValue(), fieldKey.Name)
+		for _, item := range v.fieldKeys[contextPrefixedFieldName] {
+
+			// Match items where: user didn't specify data type OR data type matches. Context filtering not needed here since context was used to construct the lookup key.
+			if fieldKey.FieldDataType == telemetrytypes.FieldDataTypeUnspecified || item.FieldDataType == fieldKey.FieldDataType {
+				fieldKeysForName = append(fieldKeysForName, item)
 			}
```
Comment (piyushsingariya) [inline pkg/querybuilder/where_clause_visitor.go]: nit: but there multiple empty newlines

File: pkg/querybuilder/where_clause_visitor_test.go [hunk 2]
```diff
@@ -73,3 +73,551 @@ func TestPrepareWhereClause_EmptyVariableList(t *testing.T) {
...
+					},
+				},
+			},
+			expectedKeys: []telemetrytypes.TelemetryFieldKey{
+				{
+					Name:          "host",
+					FieldContext:  telemetrytypes.FieldContextResource,
+					FieldDataType: telemetrytypes.FieldDataTypeString,
+				},
+			},
```
Comment (piyushsingariya) [inline pkg/querybuilder/where_clause_visitor_test.go:398]: doubt: both should've been expected right?

File: pkg/querybuilder/where_clause_visitor_test.go [hunk 3]
```diff
@@ -73,3 +73,551 @@ func TestPrepareWhereClause_EmptyVariableList(t *testing.T) {
...
+					},
+				},
+				"custom_field": {
+					{
+						Name:          "custom_field",
+						FieldContext:  telemetrytypes.FieldContextAttribute,
+						FieldDataType: telemetrytypes.FieldDataTypeString,
+					},
+				},
+			},
```
Comment (piyushsingariya) [inline pkg/querybuilder/where_clause_visitor_test.go]: doubt: since both are being considered now, how would a customer restrict the logic to not choose `attribute.custom_field`

File: pkg/querybuilder/where_clause_visitor_test.go (hunk 2 above)
Comment (tushar-signoz) [reply pkg/querybuilder/where_clause_visitor_test.go:398]: We prefer resource over attribute and throw warning

File: pkg/querybuilder/where_clause_visitor_test.go (hunk 3 above)
Comment (tushar-signoz) [reply pkg/querybuilder/where_clause_visitor_test.go]: This is a rare case but definitely possible that `attribute.count` and `count` both are valid attributes.
If we consider a case that both have same purposes, same data exists in either of columns. Then its fine even if user cannot restrict.
So we can restrict using `attribute.attribute.count` but not for the other field. Let me think on this.

Comment (tushar-signoz) [reply pkg/querybuilder/where_clause_visitor_test.go]: 1 option I can think at top of my head is `log.attribute.count`

Comment (tushar-signoz) [reply pkg/querybuilder/where_clause_visitor_test.go]: Hey, they can just say `custom_field`.

**Review by srikanthccv — APPROVED** [review]
Thanks for the visitor visit tests.
